### PR TITLE
KIALI-1543 Refactor NamespaceFilter => StatefulFilters prior to kiali- 1477 and kiali- 1528

### PR DESCRIPTION
**Pull request by jotak**
- Not dedicated to namespace anymore
- This is basically a patternfly filter that manages state
- Move some common functions related to filtering from specific list pages to common ListPage class

No visible change expected.

### CODE REVIEW

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#39](https://paste.fedoraproject.org/paste/snHAkioZ6ljjWrxBWMBjqg)
  * :red_circle: Istio-and-Kiali-on-Openshift [#[1238]](https://paste.fedoraproject.org/paste/YMDxk20XW5mcN-br0le0wQ)
  * :heavy_check_mark: Kiali-on-Openshift [#[583]](https://paste.fedoraproject.org/paste/fLU7Mbvnvdpw9PbP~mrR9Q)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1013]](https://paste.fedoraproject.org/paste/ws~hfjrgCAQ18O4jQPTr0Q)
  * :red_circle: run-kiali-e2e-tests [#[61]](https://paste.fedoraproject.org/paste/52vP7ffvlsfPiPAxHQMJHA)


KIALI_NAME=kialiuipr667-istio-system

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#40](https://paste.fedoraproject.org/paste/AYbgOOo2hUpmRxr4tMHIgQ)
  * :red_circle: Istio-and-Kiali-on-Openshift [#[1240]](https://paste.fedoraproject.org/paste/GYb1tnV9uNIWTsj3~aDOHw)
  * :heavy_check_mark: Kiali-on-Openshift [#[585]](https://paste.fedoraproject.org/paste/36Ub6QdHI8lntmnhndMF7w)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1014]](https://paste.fedoraproject.org/paste/NkWse9r3cnQInLPKDPhx0A)
  * :red_circle: run-kiali-e2e-tests [#[62]](https://paste.fedoraproject.org/paste/rgg6qhmodNwo0N3W9qlaMQ)


KIALI_NAME=kialiuipr667-istio-system

**Comment by lucasponce** [comment]
One preliminar thing I see broken on this PR.
Current logic in master is that when you select a namespace(s) in Applications, namespace(s) can be propagated to Workloads/Services/IstioConfig list pages.
But other filters should be cleaned when moving from pages as they only belongs to the page itself.

I see "App name" filter is propagated which is not correct.

cc @xeviknal (as we discussed this behaviour)

File: src/components/ListPage/ListPage.ts [hunk 1]
```diff
@@ -18,6 +19,9 @@ export namespace ListPage {
     getQueryParam: (queryName: string) => string[] | undefined;
     getSingleQueryParam: (queryName: string) => string | undefined;
     getSingleIntQueryParam: (queryName: string) => number | undefined;
+    getFiltersFromURL: (filterTypes: FilterType[]) => ActiveFilter[];
```
Comment (lucasponce) [inline src/components/ListPage/ListPage.ts]: +1

**Comment by jotak** [comment]
@lucasponce that should be good now (see last commit).

I also added some unit tests on ListPage filtering & query params management. They should help for any future change (I've been missing them! :) )

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#54](https://paste.fedoraproject.org/paste/HBPGOR0Z9UMMLZ0C2vaPIQ)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1274]](https://paste.fedoraproject.org/paste/~5ZH~BgZuuwYsdcxjStonQ)
  * :heavy_check_mark: Istio-Upstream [#[159]](https://paste.fedoraproject.org/paste/Xec~M3VhqnFeibCNU25q-w)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1064, 1065]](https://paste.fedoraproject.org/paste/BQeNnIr8YH~kqZCI1h4uIQ)
  * :heavy_check_mark: install-bookinfo [#[624]](https://paste.fedoraproject.org/paste/qY7l2md8sCiTYYeFUh3l~g)
  * :heavy_check_mark: Kiali-on-Openshift [#[621]](https://paste.fedoraproject.org/paste/LGh9fgY85DxSxG4ven~bNA)
  * :heavy_check_mark: run-kiali-e2e-tests [#[91]](https://paste.fedoraproject.org/paste/ddgVzDV-3JDomiMyn00KOg)


KIALI_NAME=kialiuipr667-istio-system

**Comment by jotak** [comment]
thanks @lucasponce

**Pull request merged**